src/OrchardCore.Modules/OrchardCore.ContentLocalization/Indexing/LocalizationPartIndexHandler.cs

```diff
@@ -0,0 +1,22 @@
+using System.Threading.Tasks;
+using OrchardCore.ContentLocalization.Models;
+using OrchardCore.Indexing;
+
+namespace OrchardCore.ContentLocalization.Indexing
+{
+    public class LocalizationPartIndexHandler : ContentPartIndexHandler<LocalizationPart>
+    {
+        public override Task BuildIndexAsync(LocalizationPart part, BuildPartIndexContext context)
+        {
+            var options = DocumentIndexOptions.Store;
+
+            foreach (var key in context.Keys)
+            {
+                context.DocumentIndex.Set(key, part.LocalizationSet, options);
```

Comment:
They both need to be stored. You can force it in the options object. Because I assume we should be able to read any of these values if we filter by the other one.